homeassistant/components/comfoconnect/config_flow.py

```diff
@@ -0,0 +1,158 @@
+            # pycomfoconnect raises
+            except Exception:  # pylint: disable=broad-except
+                errors["base"] = "unknown"
+            if unique_id:
+                await self.async_set_unique_id(unique_id)
+                self._abort_if_unique_id_configured()
+                user_input[CONF_TOKEN] = secrets.token_hex(16)
+                user_input[CONF_USER_AGENT] = DEFAULT_USER_AGENT
+                self.context[STEP_ID_USER] = user_input
+                return await self.async_step_sensor_selection()
+        return self.async_show_form(
+            step_id=STEP_ID_USER, data_schema=schema, errors=errors
+        )
+
+    async def async_step_sensor_selection(self, user_input=None):
+        """Handle sensor selection step."""
+        schema = vol.Schema(
+            {
+                vol.Optional(
+                    CONF_SENSORS,
+                ): cv.multi_select({sensor.key: sensor.name for sensor in SENSOR_TYPES})
+            }
+        )
+        errors = {}
+        unique_id = self.context["unique_id"]
+        self._abort_if_unique_id_configured()
+        if self.context["source"] == config_entries.SOURCE_IMPORT:
+            user_input = {"sensors": self.context["import_sensors"] or []}
+        if user_input is not None:
+            data = self.context[STEP_ID_USER]
+            return self.async_create_entry(
+                title=f"ComfoAir {unique_id}",
+                data=data,
+                options={"sensors": user_input["sensors"]},
+            )
+        return self.async_show_form(
+            step_id="sensor_selection",
+            data_schema=schema,
+            errors=errors,
+        )
+
+    async def async_step_import(self, import_config):
+        """Handle a flow import."""
+        return await self.async_step_user(import_config)
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry):
+        """Get the options flow for this handler."""
+        return OptionsFlowHandler(config_entry)
+
+
+class OptionsFlowHandler(config_entries.OptionsFlow):
+    """Handle ComfoConnect options."""
+
+    def __init__(self, config_entry: config_entries.ConfigEntry) -> None:
+        """Initialize ComfoConnect options flow."""
+        self.config_entry = config_entry
+
+    async def async_step_init(self, user_input=None):
+        """Manage the options."""
+        if user_input is not None:
+            return self.async_create_entry(title="", data=user_input)
+
+        return self.async_show_form(
+            step_id=STEP_ID_INIT,
+            data_schema=vol.Schema(
+                {
+                    vol.Optional(
+                        CONF_SENSORS,
+                        default=self.config_entry.options.get(CONF_SENSORS),
+                    ): cv.multi_select(
+                        {sensor.key: sensor.name for sensor in SENSOR_TYPES}
+                    )
+                }
+            ),
```

Comment:
Turning on/off entities/sensors is a Home Assistant default feature, we do not need a new construct/option flow for this.